### PR TITLE
Reliability improvements for service discovery

### PR DESCRIPTION
**Pull request by bart-vmware**
## Description

- Fixed: `ObjectDisposedException` within Consul when using discovery-first config server (service container is disposed too soon)
- Enforce fetching registry in ConfigServer discovery-first with Eureka
- Allow non-conflicting Eureka instance configuration changes, since InstanceId is already made unique when dynamic ports assignment is used (this removes the requirement to explicitly configure the Instance ID)
- Skip over throwing `IDiscoveryClient`s in load balancers (renamed `CachingServiceInstancesResolver` to `ServiceInstancesResolver`, changed static to instance and made public)
- Fixed: Do not log exception at shutdown when Eureka is disabled
- Fixed: do not report health when Consul is disabled

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [x] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [ ] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

File: build/pr-code-cleanup.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ jobs:
 
         if ($baseCommitHash -ne $headCommitHash) {
           Write-Output "Running code cleanup on commit range $baseCommitHash..$headCommitHash in pull request."
-          dotnet regitlint -s src/Steeltoe.All.sln --print-command --skip-tool-check --jb --dotnetcoresdk=$(dotnet --version) --jb-profile="Steeltoe Full Cleanup" --jb --properties:Configuration=Release --jb --verbosity=WARN -f commits -a $headCommitHash -b $baseCommitHash
+          dotnet regitlint -s src/Steeltoe.All.sln --print-command --skip-tool-check --jb --dotnetcoresdk=$(dotnet --version) --jb-profile="Steeltoe Full Cleanup" --jb --properties:Configuration=Release --jb --verbosity=WARN --jb --properties:NuGetAudit=false -f commits -a $headCommitHash -b $baseCommitHash
```
Comment (TimHess) [inline build/pr-code-cleanup.yml:73]: Does this PR set NuGetAudit=false on all builds?

**Review by TimHess — DISMISSED** [review]
LGTM

File: build/pr-code-cleanup.yml (hunk 1 above)
Comment (bart-vmware) [reply build/pr-code-cleanup.yml:73]: No, only when running code cleanup.

Comment (bart-vmware) [reply build/pr-code-cleanup.yml:73]: During regular builds, a warning is logged during `dotnet restore`. Code cleanup runs that internally, while warn-as-error is true, causing its build to fail.

**Review by TimHess — DISMISSED** [review]

**Review by TimHess — APPROVED** [review]

**Pull request merged**